In [4]:
import pandas as pd
import numpy as np
import warnings
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import train_test_split
warnings.filterwarnings('ignore')



In [5]:
#Importing Clean Data.
data_frame=pd.read_csv('Encoded_Data.csv')

In [6]:
data_frame.head()

Sex   Age  Height  Weight  Team   NOC  Year  Season   City  Event  Medal
0    1  24.0   180.0    80.0  4975  5141  1992       0  12977   3280      0
1    1  23.0   170.0    60.0  4975  5141  2012       0  22426    341      0
2    1  24.0   175.0    70.0  3424  3570  1920       0   4292   5733      0
3    1  34.0   175.0    70.0     6  3570  1900       0   7169    170      1
4    0  21.0   185.0    82.0  5718  5839  1988       1   2639    467      0

In [43]:
#Seperating Input and output Labels.
X = data_frame.drop('Medal',axis=1)
Y = data_frame[['Medal']]

# Spliting Data

In [44]:
x_train, x_test, y_train, y_test = train_test_split(X, Y, test_size = 0.33, random_state = 100)

In [45]:
print("X_train shape:", x_train.shape) 
print("X_test shape:", x_test.shape) 
print("Y_train shape:", y_train.shape) 
print("Y_test shape:", y_test.shape)

X_train shape: (181647, 10)
X_test shape: (89469, 10)
Y_train shape: (181647, 1)
Y_test shape: (89469, 1)


# XgBoosting and Adaboost Classifiers

The XGBoost(Extreme Gradient Boosting) algorithm uses decision tree machine learning algorithm and boosting process which increases the  performance.

In [46]:
import xgboost as xgb
from sklearn.datasets import load_wine
from sklearn.metrics import accuracy_score
from sklearn.metrics import classification_report
from xgboost import XGBClassifier


In [47]:
model = XGBClassifier(use_label_encoder=False, eval_metric='mlogloss')

In [48]:
#Training model on train data.
model.fit(x_train, y_train)

XGBClassifier(eval_metric='mlogloss', objective='multi:softprob',
              use_label_encoder=False)

In [26]:
#Predicting values for input test data
y_pred = model.predict(x_test)

In [49]:
accuracy = accuracy_score(y_test, y_pred)
accuracy

0.8547206294917793

From the Accuracy score we can assume that it is an efficient model for our dataset, but we should aslo see classification report for clear insights.

In [29]:
print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       0.85      1.00      0.92     76435
           1       0.71      0.01      0.01      4401
           2       0.42      0.00      0.00      4361
           3       0.38      0.00      0.00      4272

    accuracy                           0.85     89469
   macro avg       0.59      0.25      0.24     89469
weighted avg       0.80      0.85      0.79     89469



1) From the classification report we can see that precision,recall,f1-score for 0 have same values as logistic regression. 
2) Logistic regression failed to predict 1,2 and 3, after applying Xg boost algorithm we can see the model giving decent prediction percentages for values other than 0.
3) Even though Xg boost model's acuracy score is 85% it produced less recall and precisions scores for 1,2 and 3 when compared with KNN model.
4) Considering all these scenarios we can say that we need an algorithm which can produce decent recall values with good prediction percentages.
